Training

jburel opened the pull request:
# What this PR does

Reorganise the examples so they can be included in the daily build
Fix unit

Other examples have been reviewed as part of https://github.com/openmicroscopy/openmicroscopy/pull/4914

jburel committed:
Adjust script so it can be included in daily build

jburel committed:
Rework code so it can be included

jburel committed:
Rework script so it can be included in build

jburel committed:
Rename file to match convention

jburel committed:
Rework so it can be included in build

jburel committed:
Fix units issue and reorganize so it can be run daily

jburel committed:
Include more scripts

jburel committed:
flake8

will-moore commented on examples/Training/python/Advanced/Create_Image_advanced.py:
```diff
@@ -24,8 +28,11 @@
 
 # Configuration
```
This is no-longer `#Configuration` here but something like
" Find another image by getting first image in Dataset"

will-moore commented on examples/Training/python/Advanced/Create_Image_advanced.py:
```diff
@@ -89,17 +96,11 @@ def planeGen():
-    newPix.setPhysicalSizeZ(rdouble(physicalSizeZ))
+newPix.setPhysicalSizeZ(pixels.getPhysicalSizeZ())
 
 conn.getUpdateService().saveObject(newPix)
```
Can probably remove empty lines above, making this all one block, up to `newPix = ...`

sbesson commented on examples/Training/python/Advanced/Create_Image_advanced.py:
```diff
@@ -7,14 +7,18 @@
 
+import sys
+import os
+sys.path.append(os.path.join('..', 'python'))
```
Two thoughts to prevent the Python path workaround assuming it is in place for importing `Parse_OMERO_Properties`:
- either use relative imports e.g. `from ..Parse_OMERO_Properties import...`
- or maybe migrate all the `Advanced` examples together with the rest of the other examples

jburel replied:
`from ..Parse_OMERO_Properties import...` does not work with the actual set-up

will-moore commented:
The `Task_Scripts/Write_Data_3.py` and `Task_Scripts/Write_Data_4.py` are meant to be sample OMERO.scripts that can be used as starting points for writing your own scripts. But now these are not OMERO.scripts any more. Either they need to be left as they are or moved somewhere else if we don't want un-testable code there. Best solution would be to actually upload them via CLI and run them as part of the test, but that would be a fair bit harder?

jburel commented:
We already have some "samples" scripts under "example/ScriptingService" (that are not tested!)
Let's discuss where we want them.
The main problem here is that they are not run or tested on a regular basis and some have not been updated since the units work so wrong examples!
Either we test them or we delete them not much choice in my opinion

jburel committed:
Delete scripts

Both can already be found https://github.com/openmicroscopy/omero-example-scripts

jburel committed:
Remove script

Can be found https://github.com/openmicroscopy/omero-example-scripts

jburel committed:
Remove scripts from the list

jburel commented:
currently the "training" user (who "runs" the examples) is not an admin so we can't upload as official then run the script to test its contents
Better to have a general testing infrastructure for https://github.com/openmicroscopy/omero-example-scripts

jburel committed:
Minor edits

jburel committed:
Remove python path work-around

jburel committed:
Swap image to copy

jburel committed:
Flake8

jburel committed:
Use the fake image imported during set up

jburel committed:
Change order

jburel commented:
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-training/446/ is green again

will-moore commented on examples/Training/python/Advanced/Create_Image_advanced.py:
```diff
@@ -11,30 +11,38 @@
-
-# Configuration
+print HOST
+print PORT
```
Is there a reason to have these prints, or are they just left-over from debugging? Bit cleaner without them.

jburel replied:
I was debugging.
I will remove them

will-moore commented:
Output from https://ci.openmicroscopy.org/job/OMERO-DEV-merge-training/446/ looks good.
NB: I noticed a bunch of Exceptions from the Java examples, but they have been there for a while.

One more thought: The great thing about having the Training examples running is that the same code is used on the docs pages https://www.openmicroscopy.org/site/support/omero5.2/developers/Python.html etc. 
The `Training/markup.py` script should do this, but it's getting fragile (output from script is deviating more and more from what's on the docs pages.
Would be nice to try and port the changes here to the Python docs page where appropriate.

jburel commented:
Wrt to doc
Certainly to look at but I won't do it in this PR
I will open a card

jburel committed:
Remove print introduced while debugging

jburel commented:
@will-moore: are you okay with the last commit f6427ca?
Travis is now green

so I can integrate and activate the remaining examples as part of https://github.com/openmicroscopy/openmicroscopy/pull/4914

will-moore commented:
Yes, all good, thanks

jburel commented:
Merging so I can enable the other examples
